lib/Dialect/HW/HWOps.cpp

```diff
@@ -407,6 +407,11 @@ static LogicalResult checkAttributes(Operation *op, Attribute attr, Type type) {
     if (!arrayAttr)
       return op->emitOpError("expected array attribute for constant of type ")
              << type;
+    if (structType.getElements().size() != arrayAttr.size()) {
+      return op->emitOpError("array attribute (")
+             << arrayAttr.size() << ") has wrong size for struct constant ("
+             << structType.getElements().size() << ")";
+    }
```

Comment:
nit: please omit braces for single line statements
```suggestion
    if (structType.getElements().size() != arrayAttr.size())
      return op->emitOpError("array attribute (")
             << arrayAttr.size() << ") has wrong size for struct constant ("
             << structType.getElements().size() << ")";

```